src/modules/rtpengine/rtpengine.c

```diff
@@ -3323,9 +3341,18 @@ static bencode_item_t *rtpp_function_call(bencode_buffer_t *bencbuf,
 	}
 
 	/* initialize some basic bencode items */
-	ng_flags.dict = bencode_dictionary(bencbuf);
+	if(!extra_dict) {
+		ng_flags.dict = bencode_dictionary(bencbuf);
+		if(parse_by_module) {
+			ng_flags.flags = bencode_list(bencbuf);
+		}
+	} else {
+		ng_flags.dict = extra_dict;
+		ng_flags.flags = bencode_dictionary_get(ng_flags.dict, "flags");
```

Comment:
Do you really need to create the `flags` here? Aren't the calling functions creating it?

Reply:
Yes I need. Before this change, it was like:
```
if(parse_by_module) {
		ng_flags.flags = bencode_list(bencbuf);
		ng_flags.received_from = bencode_list(bencbuf);
	}
```
but know I have some flags in extra_dict, so I need to create flags in ng_flags from flags in extra_dict.